src/Raven.Client/Documents/Operations/Revisions/GetRevisionsOperation.cs

```diff
@@ -0,0 +1,77 @@
+//-----------------------------------------------------------------------
+// <copyright file="RevisionsConfiguration.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Net.Http;
+using Raven.Client.Documents.Commands;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Extensions;
+using Raven.Client.Http;
+using Raven.Client.Json.Serialization.NewtonsoftJson.Internal;
+using Sparrow.Json;
+
+namespace Raven.Client.Documents.Operations.Revisions
+{
+    public class GetRevisionsOperation<T> : IOperation<RevisionsResult<T>>
+    {
+        private readonly GetRevisionsResultCommand _command;
+
+        public GetRevisionsOperation(string id, int? start = 0, int? pageSize = int.MaxValue, bool metadataOnly = false)
```

Comment:
create two ctors:
1. (string id, int? start = 0, int? pageSize = int.MaxValue) - for convenience
2. (Parameters parameters) - nested class - check other operations how this is done

Introduce validation of the parameters e.g. can id be null?

Reply:
Why both default params and nullables?